include/envoy/http/header_map.h

```diff
@@ -317,6 +317,7 @@ class HeaderEntry {
   HEADER_FUNC(GrpcStatus)                                                                          \
   HEADER_FUNC(GrpcTimeout)                                                                         \
   HEADER_FUNC(Host)                                                                                \
+  HEADER_FUNC(Http2Settings)                                                                       \
```

Comment:
Given how much of a corner case this is my suggestion was to not waste an O(1) slot here and just remove it by name directly for now. WDYT?

Reply:
Sure. Done.